RecoParticleFlow/PFProducer/plugins/MLPFProducer.cc

```diff
@@ -0,0 +1,139 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+#include "RecoParticleFlow/PFProducer/interface/MLPFModel.h"
+
+struct MLPFCache {
+  std::atomic<tensorflow::GraphDef*> graph_def;
```

Comment:
Why is this atomic? From the usage point of view it looks like it should really be `mutable` (and even that could be avoided if
https://github.com/cms-sw/cmssw/blob/f4f4ada6ccb7fb09381e05a77367aacee217c321/PhysicsTools/TensorFlow/interface/TensorFlow.h#L96
would take `GraphDef const*`, which it could, IIUC)

Reply:
right - thank you for pointing it out! I think at the time I wrote this (~6mo ago), the example I used was based on atomic, I think our understanding of TF thread safety guarantees has evolved since. I will try mutable.